### PR TITLE
Nano: improve Howto guide of InferenceOptimizer.optimize

### PR DESCRIPTION
**Pull request by rnwang04**
## Description

Update How-to guide of `InferenceOptimizer.optimize`.

### 1. Why the change?

 As we have greatly improved `InferenceOptimizer.optimize` method by extending the input format / extending the combinations of acceleration methods / updating output format / providing more functions or modes for user, we need to introduce these usages to users by How-to guide.

### 2. User API changes

No change.

### 3. Summary of the change 

- [x] update output table of `InferenceOptimizer.optimize`
- [x] add how-to guide for different mode : default / all
- [x] add how-to guide for filtering acceleration methods
- [x] add how-to guide for disable validation
- [x] add how-to guide for Tensor input
- [x] add how-to guide for `get_model`
- [x] update description of exported model

### 4. How to test?
- [x] Unit test
- [x] Document test
https://ruonantetdoc.readthedocs.io/en/update_howto_optimizer/doc/Nano/Howto/Inference/PyTorch/inference_optimizer_optimize.html

### CODE REVIEW

File: python/nano/tutorial/notebook/inference/pytorch/inference_optimizer_optimize.ipynb [hunk 1]
```diff
@@ -187,26 +196,24 @@
...
+    "|  jit_fp32_ipex_channels_last   |      successful      |    21.505    |        0.794*        |\n",
+    "|         jit_bf16_ipex          |      successful      |     9.7      |        0.792         |\n",
+    "|  jit_bf16_ipex_channels_last   |      successful      |     9.84     |        0.792         |\n",
+    "|         openvino_fp32          |      successful      |    24.205    |        0.794*        |\n",
+    "|         openvino_int8          |      successful      |    5.805     |        0.792         |\n",
+    "|        onnxruntime_fp32        |      successful      |    19.792    |        0.794*        |\n",
+    "|    onnxruntime_int8_qlinear    |      successful      |     7.34     |         0.79         |\n",
     " -------------------------------- ---------------------- -------------- ----------------------\n",
-    "Optimization cost 74.2s in total.\n",
+    "* means we assume the precision of the traced model does not change, so we don't recompute accuracy to save time.\n",
```
Comment (MeouSker77) [inline python/nano/tutorial/notebook/inference/pytorch/inference_optimizer_optimize.ipynb:215]: may change 'precision' to 'accuracy' ?

File: python/nano/tutorial/notebook/inference/pytorch/inference_optimizer_optimize.ipynb [hunk 2]
```diff
@@ -222,17 +229,302 @@
...
+    "Optimization cost 53.2s in total.\n",
+    "```"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 4. Disable validation during optimize\n",
```
Comment (MeouSker77) [inline python/nano/tutorial/notebook/inference/pytorch/inference_optimizer_optimize.ipynb:451]: change 'optimize' to `optimizing` or `optimization` ?

File: python/nano/tutorial/notebook/inference/pytorch/inference_optimizer_optimize.ipynb [hunk 3]
```diff
@@ -222,17 +229,302 @@
...
+    "\n",
+    "If you can't get corresponding validation dataloader for you model, or you don't care about the possible accuracy drop, you could just disable validation by don't specify `validation_data`, `metric`, `direction` paramater:"
```
Comment (MeouSker77) [inline python/nano/tutorial/notebook/inference/pytorch/inference_optimizer_optimize.ipynb:453]: change 'you could just disable validation by don't specify `validation_data`, `metric`, `direction` paramater:' to 'you can omit `validataion_data`, `metrice` and `direction` parameters to disable validation'

File: python/nano/tutorial/notebook/inference/pytorch/inference_optimizer_optimize.ipynb [hunk 4]
```diff
@@ -222,17 +229,302 @@
...
+    "                   training_data=train_dataloader,\n",
+    "                   thread_num=1,\n",
+    "                   latency_sample_num=100)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 5. More flexiable input format\n",
```
Comment (MeouSker77) [inline python/nano/tutorial/notebook/inference/pytorch/inference_optimizer_optimize.ipynb:472]: 'flexiable' to 'flexible'

Comment (rnwang04) [reply python/nano/tutorial/notebook/inference/pytorch/inference_optimizer_optimize.ipynb]: done

File: python/nano/tutorial/notebook/inference/pytorch/inference_optimizer_optimize.ipynb (hunk 3 above)
Comment (rnwang04) [reply python/nano/tutorial/notebook/inference/pytorch/inference_optimizer_optimize.ipynb]: done

File: python/nano/tutorial/notebook/inference/pytorch/inference_optimizer_optimize.ipynb (hunk 1 above)
Comment (rnwang04) [reply python/nano/tutorial/notebook/inference/pytorch/inference_optimizer_optimize.ipynb]: done, have revised to `accuracy`

**Comment by TheaperDeng** [comment]
This might be a recommendation we could do in next PR. We may reorganize this how to guide to

1. divide the whole guide to 2 parts
   
   - Use `InferenceOptimizer` to optimize the model with best optimization method. This could contains
     
     - User default search mode to call `inferenceoptimizer.optimze`
     
     - Get best optimized model by `inferenceoptimizer.get_best_model`
     
     - Use the optimized model to do some inference
   
   - **[Most people will get what they want when they have read till here]**
   
   - Other advanced functions
     
     - Apply more precise searching space among optimization method (describe search mode)
     
     - Get a optimized model by specific method (describe `inferenceoptimizer.get_model`)
     
     - Export/load the optimized model (decribe `InferenceOptimier.save/load`)

**Comment by rnwang04** [comment]
> This might be a recommendation we could do in next PR. We may reorganize this how to guide to
> 
> 1. divide the whole guide to 2 parts
>    
>    * Use `InferenceOptimizer` to optimize the model with best optimization method. This could contains
>      
>      * User default search mode to call `inferenceoptimizer.optimze`
>      * Get best optimized model by `inferenceoptimizer.get_best_model`
>      * Use the optimized model to do some inference
>    * **[Most people will get what they want when they have read till here]**
>    * Other advanced functions
>      
>      * Apply more precise searching space among optimization method (describe search mode)
>      * Use Tensor input sample rather than dataloader
>      * Get a optimized model by specific method (describe `inferenceoptimizer.get_model`)
>      * Export/load the optimized model (decribe `InferenceOptimier.save/load`)

will update in next PR.
